src/OracleService/OracleService.cs

```diff
@@ -86,6 +87,10 @@ private void OnStart()
             {
                 if (!CheckOracleAvaiblable(snapshot, out ECPoint[] oracles)) throw new ArgumentException("The oracle service is unavailable");
                 if (!CheckOracleAccount(wallet, oracles)) throw new ArgumentException("There is no oracle account in wallet");
+                foreach (var (_, p) in protocols)
+                {
+                    if (p is OracleFsProtocol) ((OracleFsProtocol)p).AttachWallet(wallet, oracles);
```

Comment:
Why it needs wallet?

Reply:
@ZhangTao1596  Can you describe it in detail？

Reply:
All NeoFS requests need to be signed for security purposes and for ACL functionality. If we expect oracles to only access publicly-available containers (not restricted by ACLs) then this key can be autogenerated during startup, but still a client needs some key. At the same time it's better to have something static here as in the future this key might be checked with NeoID.

Reply:
So every neofs node needs a wallet? It can't connect to the network without a wallet?

Reply:
> So every neofs node needs a wallet? It can't connect to the network without a wallet?

Yes it can't. Every NeoFS node and client needs a public key to identify itself and private key to sign messages during p2p communication with other nodes. As the key format is the same for Neo and NeoFS, it can use wallet for it.

As for accessing public data, client can even generate random key for each request, it is valid option.